src/components/analytics/ComparisonAnalysis.vue

```diff
@@ -11,178 +11,296 @@
 
     return {
+      chartData: {},
       startDate: "2022-01-01",
       endDate: "2022-05-20",
       company: 0,
+      company_type: "",
       companies_types: [
-        {name: 'FACTORY', value:'factory'},
-        {name: 'STORE', value:'store'},
-        {name: 'RETIAL', value:'retial'},
-        {name: 'VENDOR', value:'vendor'},
-        {name: 'PARENT', value:'parent'},
-        {name: 'WALK_IN_CUSTOMER', value:'walk_in_customer'},
-      ]
+        { name: "FACTORY", value: "factory" },
+        { name: "STORE", value: "store" },
+        { name: "RETIAL", value: "retial" },
+        { name: "VENDOR", value: "vendor" },
+        { name: "PARENT", value: "parent" },
+        { name: "WALK_IN_CUSTOMER", value: "walk_in_customer" },
+      ],
+      printDelay: 100,
     };
   },
   computed: {
     ...mapGetters({
-      salesanalytics: 'getSalesanalytics',
-      // companies: 'getInventoryCompanies',
-      userdata: 'getUser',
+      salesanalytics: "getSalesanalytics",
+      userdata: "getUser",
+      comparisonAnalysis: "getComparisonanalysis",
     }),
   },
-  methods: { 
-    admin(){
-      const allowedRoles = ['SUPER_ADMIN', 'ADMIN'];
-      const allowedCompanies = ['PARENT', 'STORE'];
-      if(this.userdata != null && allowedRoles.includes(this.userdata.user_type)
-        && allowedCompanies.includes(this.userdata.company.company_type) 
-      ){
+  methods: {
+    ...mapActions({
+      getComparisonAnalysis: ActionTypes.FETCH_COMPARISON_ANALYSIS,
+    }),
+    admin() {
+      const allowedRoles = ["SUPER_ADMIN", "ADMIN"];
+      const allowedCompanies = ["PARENT", "STORE"];
+      if (
+        this.userdata != null &&
+        allowedRoles.includes(this.userdata.user_type) &&
+        allowedCompanies.includes(this.userdata.company.company_type)
+      ) {
         return true;
       }
       return false;
     },
+    async fetchAnalyticsComparisonBtn() {
+      await this.getComparisonAnalysis({
+        company_type: this.company,
+      });
+      this.chartData = {
+        labels: [],
+        datasets: [
+          {
+            label: "Companies",
+            data: JSON.parse(
+              JSON.stringify(store.getters.getComparisonanalysis)
+            ),
+            backgroundColor: [
+              "rgba(255, 99, 132, 0.2)",
+              "rgba(255, 159, 64, 0.2)",
+              "rgba(255, 205, 86, 0.2)",
+              "rgba(75, 192, 192, 0.2)",
+              "rgba(54, 162, 235, 0.2)",
+              "rgba(153, 102, 255, 0.2)",
+              "rgba(201, 203, 207, 0.2)",
+            ],
+            borderColor: [
+              "rgb(255, 99, 132)",
+              "rgb(255, 159, 64)",
+              "rgb(255, 205, 86)",
+              "rgb(75, 192, 192)",
+              "rgb(54, 162, 235)",
+              "rgb(153, 102, 255)",
+              "rgb(201, 203, 207)",
+            ],
+          },
+        ],
+      };
+    },
+    getElementTag: function(tag: keyof HTMLElementTagNameMap): string {
```

Comment:
move all print related functions to a separate file. 
So that they can be reused in future. There is a lot of repetition rn. 
Change definitions of methods so that they are not some component specific.